cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -720,50 +730,121 @@ function getPriceHtml(product) {
+    url.searchParams.append('objectID', objectID);
+    url.searchParams.append('indexName', indexName);
+    return url.href;
+}
+
+/**
+ * Determine final display price, strikeout price, and calloutMsg for an item
  * @param {Object} item The item object
- * @return {number} The calculated sales price
+ * @param {Object} algoliaData The Algolia data configuration
+ * @param {Array} activeCustomerPromotions The list of active promotions
+ * @returns {Object} { displayPrice, strikeoutPrice, calloutMsg }
  */
-function calculateDisplayPrice(item) {
-    var promotions;
-    var calloutMsg = '';
-    var variant;
-
-    if (algoliaData.recordModel === 'master-level' && item.variants) {
-        promotions = item.variants.length > 0 ? item.variants[0].promotions : item.promotions;
-        variant = item.variants.length > 0 ? item.variants[0] : item;
-    } else {
-        promotions = item.promotions;
-        variant = item;
+function determinePrices(item, algoliaData, activeCustomerPromos) {
+    let displayPrice = item.price;
+    let strikeoutPrice = null;
+    let calloutMsg = '';
+
+    if (item.pricebooks && item.pricebooks[algoliaData.currencyCode]) {
+        const { computedPrice } = applyPricebooks(item, algoliaData);
+        if (computedPrice !== null) {
+            if (computedPrice < displayPrice) {
+                strikeoutPrice = displayPrice; 
+                displayPrice = computedPrice; 
+            } else if (computedPrice > displayPrice) {
+                // If pricebooks show a higher original price
+                strikeoutPrice = computedPrice;
+            }
+        }
     }
 
-    if (promotions && promotions[algoliaData.currencyCode]) {
-        var productPromos = promotions[algoliaData.currencyCode];
-        var minPrice = algoliaData.recordModel === 'master-level' ? (variant.price[algoliaData.currencyCode] ? variant.price[algoliaData.currencyCode] : variant.price) : item.price;
-        for (var i = 0; i < activeCustomerPromotions.length; i++) {
-            for (var j = 0; j < productPromos.length; j++) {
-                if (productPromos[j].promoId === activeCustomerPromotions[i].id && productPromos[j].price < minPrice) {
-                    minPrice = productPromos[j].price;
-                    calloutMsg = activeCustomerPromotions[i].calloutMsg;
-                }
-            }
+    if (item.promotionalPrice && item.promotionalPrice < item.price) {
+        strikeoutPrice = item.price;
+        displayPrice = item.promotionalPrice;
+    }
+
+    if (item.promotions && item.promotions[algoliaData.currencyCode]) {
+        const { promotionalPriceFromPromos, msg } = applyPromotions(item, algoliaData, activeCustomerPromos);
+        if (promotionalPriceFromPromos < displayPrice) {
+            displayPrice = promotionalPriceFromPromos;
+            calloutMsg = msg;
         }
+    }
+
+    if (!strikeoutPrice) {
+        strikeoutPrice = item.price > displayPrice ? item.price : null;
```

Comment:
This can be done immediately in the condition above, after retrieving the promotions (line 780-782).